puree/src/main/java/com/cookpad/puree/storage/PureeSQLiteStorage.java

```diff
@@ -110,6 +122,28 @@ public void delete(Records records) {
     }
 
     @Override
+    public void truncateBufferedLogs(int maxRecords) {
+        int recordSize = getRecordCount();
+        if (recordSize > maxRecords) {
+            String query = "SELECT * FROM " + TABLE_NAME +
```

Comment:
Can we use `DELETE FROM $table_name WHERE id IN (SELECT id FROM $table_name ORDER BY id DESC LIMIT $number_to_remove;` ? It might be faster than this PR.